security-oauth2-quickstart/src/test/java/org/acme/security/oauth2/TokenSecuredResourceTest.java

```diff
@@ -0,0 +1,76 @@
+package org.acme.security.oauth2;
+
+import io.quarkus.test.junit.QuarkusTest;
+import io.restassured.RestAssured;
+import org.junit.ClassRule;
+import org.junit.jupiter.api.Test;
+import org.testcontainers.containers.BindMode;
+import org.testcontainers.containers.FixedHostPortGenericContainer;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+@Testcontainers
+@QuarkusTest
+public class TokenSecuredResourceTest {
+
+    private static final String KEYCLOAK_SERVER_URL = System.getProperty("keycloak.url", "http://localhost:8180/auth");
+    private static final String KEYCLOAK_REALM = "quarkus";
+
+    @ClassRule
+    @Container
+    private static GenericContainer keycloak = new FixedHostPortGenericContainer("quay.io/keycloak/keycloak:9.0.3")
```

Comment:
This should be moved to a Quarkus Test Resources, and probably bound to a free port. 

Also, I'm surprised about @ClassRule as it's a Junit 4 annotation.

Reply:
Okey I will move try to move this and try to bound to a free port.

Yes you are correct ``@ClassRule`` is a Junit4 annotation. I misread the testcontainer documentation. I will be removing this in general.